From 644faecc3215f2de9b412a6f9adce362d2c9c917 Mon Sep 17 00:00:00 2001
From: sourabhsisodia_ <sourabhsisodia010@gmail.com>
Date: Mon, 16 Dec 2024 14:58:34 +0530
Subject: [PATCH 1/2] 3492645-aria-label-overrides: Fix the aria label
 implementation.

---
 js/extlink.js | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/js/extlink.js b/js/extlink.js
index 43c2b27..5a76e6f 100644
--- a/js/extlink.js
+++ b/js/extlink.js
@@ -425,19 +425,26 @@
         if (drupalSettings.data.extlink.extUseFontAwesome) {
           iconElement = document.createElement('span');
           iconElement.setAttribute('class', `fa-${className} extlink`);
+
+          // Set aria attributes on the icon element, not the link
+          iconElement.setAttribute('role', 'img');
+          iconElement.setAttribute('aria-hidden', drupalSettings.data.extlink.extHideIcons);
+
           if (className === drupalSettings.data.extlink.mailtoClass) {
             if (drupalSettings.data.extlink.mailtoLabel) {
-              link.ariaLabel = drupalSettings.data.extlink.mailtoLabel;
+              iconElement.setAttribute('aria-label', drupalSettings.data.extlink.mailtoLabel);
             }
             iconElement.innerHTML = Drupal.theme('extlink_fa_mailto', drupalSettings, iconPlacement);
-          } else if (className === drupalSettings.data.extlink.extClass) {
+          }
+          else if (className === drupalSettings.data.extlink.extClass) {
             if (drupalSettings.data.extlink.extLabel) {
-              link.ariaLabel = drupalSettings.data.extlink.extLabel;
+              iconElement.setAttribute('aria-label', drupalSettings.data.extlink.extLabel);
             }
             iconElement.innerHTML = Drupal.theme('extlink_fa_extlink', drupalSettings, iconPlacement);
-          } else if (className === drupalSettings.data.extlink.telClass) {
+          }
+          else if (className === drupalSettings.data.extlink.telClass) {
             if (drupalSettings.data.extlink.telLabel) {
-              link.ariaLabel = drupalSettings.data.extlink.telLabel;
+              iconElement.setAttribute('aria-label', drupalSettings.data.extlink.telLabel);
             }
             iconElement.innerHTML = Drupal.theme('extlink_fa_tel', drupalSettings, iconPlacement);
           }
-- 
GitLab


From 48da6a78d5cc5fceb26902888888e8cf4c47053b Mon Sep 17 00:00:00 2001
From: sourabhsisodia_ <sourabhsisodia010@gmail.com>
Date: Mon, 16 Dec 2024 15:05:19 +0530
Subject: [PATCH 2/2] 3492645-aria-label-overrides: Fix formatting issue.

---
 js/extlink.js | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/js/extlink.js b/js/extlink.js
index 5a76e6f..2910341 100644
--- a/js/extlink.js
+++ b/js/extlink.js
@@ -435,14 +435,12 @@
               iconElement.setAttribute('aria-label', drupalSettings.data.extlink.mailtoLabel);
             }
             iconElement.innerHTML = Drupal.theme('extlink_fa_mailto', drupalSettings, iconPlacement);
-          }
-          else if (className === drupalSettings.data.extlink.extClass) {
+          } else if (className === drupalSettings.data.extlink.extClass) {
             if (drupalSettings.data.extlink.extLabel) {
               iconElement.setAttribute('aria-label', drupalSettings.data.extlink.extLabel);
             }
             iconElement.innerHTML = Drupal.theme('extlink_fa_extlink', drupalSettings, iconPlacement);
-          }
-          else if (className === drupalSettings.data.extlink.telClass) {
+          } else if (className === drupalSettings.data.extlink.telClass) {
             if (drupalSettings.data.extlink.telLabel) {
               iconElement.setAttribute('aria-label', drupalSettings.data.extlink.telLabel);
             }
-- 
GitLab

