diff --git a/modules/contrib/group/src/Entity/Form/GroupContentDeleteForm.php b/modules/contrib/group/src/Entity/Form/GroupContentDeleteForm.php
index 6359c65b2..1a44279df 100644
--- a/modules/contrib/group/src/Entity/Form/GroupContentDeleteForm.php
+++ b/modules/contrib/group/src/Entity/Form/GroupContentDeleteForm.php
@@ -27,7 +27,7 @@ protected function getContentPlugin() {
    * {@inheritdoc}
    */
   public function getQuestion() {
-    return $this->t('Are you sure you want to delete %name?', ['%name' => $this->entity->label()]);
+    return $this->t('Are you sure you want to remove %name?', ['%name' => $this->entity->label()]);
   }
 
   /**
