diff --git a/src/ConditionalFieldsFormHelper.php b/src/ConditionalFieldsFormHelper.php
index 2a7d082ffc416a4dbab7a73b7d80e70ab9eb6e75..ff152c455a27461e94e3fd4b2b34357ae737d036 100644
--- a/src/ConditionalFieldsFormHelper.php
+++ b/src/ConditionalFieldsFormHelper.php
@@ -633,10 +633,6 @@ class ConditionalFieldsFormHelper {
       return;
     }
 
-    // Mark submitted values for removal. We have to remove them after all
-    // fields have been validated to avoid collision between dependencies.
-    $form_state_addition['parents'] = $dependent['field_parents'];
-
     // Optional behavior: reset the field to its default values.
     // Default values are always valid, so it's safe to skip validation.
     if (!empty($element['#conditional_fields_reset_if_untriggered']) && !$triggered) {
@@ -649,6 +645,7 @@ class ConditionalFieldsFormHelper {
     // Store dependent to remove validation errors previously set.
     // See: ConditionalFieldsFormHelper::formValidate().
     $form_state_addition['dependent'] = $dependent;
+    if ($inline_entity_form_parents) $form_state_addition['inline_entity_form_parents'] = $inline_entity_form_parents;
 
     // Tag validation errors previously set on this field for removal in
     // ConditionalFieldsFormHelper::formValidate().
@@ -803,37 +800,30 @@ class ConditionalFieldsFormHelper {
     $untriggered_dependents_errors = [];
     $form_state_errors = $form_state->getErrors();
 
-    // Tag errors from untriggered fields to remove them later.
     $untriggered_dependents = $form_state->getValue('conditional_fields_untriggered_dependents');
-    foreach ($untriggered_dependents as &$field) {
-      $error_key_arr = NestedArray::getValue($form, $field['parents'])['#parents'] ?? [];
-      $error_key = implode('][', $error_key_arr ?? []);
-      if ($error_key && isset($form_state_errors[$error_key])) {
-        $field['errors'][$error_key] = $form_state_errors[$error_key];
-      }
-    }
-
     foreach ($untriggered_dependents as $field) {
+      $inline_entity_form_parents = $field['inline_entity_form_parents'] ?? null;
       // When there is a subform, the parent location is different.
       $subform_location = \array_search('subform', $field['dependent']['array_parents'], TRUE);
       if ($subform_location !== FALSE) {
         $dependent = NestedArray::getValue($form, $field['dependent']['array_parents']);
         $parents = $dependent['#parents'];
         $field_name = end($parents);
-        $route_to_parent = array_slice($dependent['#parents'], 0, -1);
-        $field_values_location = NestedArray::getValue($form_state->getValues(), $route_to_parent);
-        $dependent_parents = $dependent['#parents'];
+        $dependent_parents = array_slice($dependent['#parents'], 0, -1);
+        $field_values_location = NestedArray::getValue($form_state->getValues(), $dependent_parents);
       }
       else {
-        $parent = [$field['parents'][0]];
-        $dependent = NestedArray::getValue($form, $parent);
-        $field_values_location = self::formFieldGetValues($dependent, $form_state);
-
-        $route_to_parent = array_slice($dependent['#parents'], 0, -1);
-        $field_values_location = NestedArray::getValue($form_state->getValues(), $route_to_parent);
+        $dependent = NestedArray::getValue($form, [...$inline_entity_form_parents['element_parents'] ?? [], $field['dependent']['field_parents'][0]]);
+        $dependent_parents = array_slice($dependent['#parents'], 0, -1);
+        $field_values_location = NestedArray::getValue($form_state->getValues(), $dependent_parents);
+        $field_name = end($dependent['#array_parents']);
+      }
 
-        $dependent_parents = $dependent['#parents'];
-        $field_name = reset($dependent['#array_parents']);
+      // Tag errors from untriggered fields to remove them later.
+      $error_key_arr = [...$dependent_parents, $field_name];
+      $error_key = implode('][', $error_key_arr ?? []);
+      if ($error_key && isset($form_state_errors[$error_key])) {
+        $field['errors'][$error_key] = $form_state_errors[$error_key];
       }
 
       // If we couldn't find a location for the field's submitted values, let
@@ -845,13 +835,21 @@ class ConditionalFieldsFormHelper {
         continue;
       }
 
-      // Save the changed array back in place.
-      // Do not use form_set_value() since it assumes
-      // that the values are located at
-      // $form_state['values'][ ... $element['#parents'] ... ], while the
-      // documentation of hook_field_widget_form() states that field values are
-      // $form_state['values'][ ... $element['#field_parents'] ... ].
-      NestedArray::setValue($form_state->getValues(), $dependent_parents, $field_values_location[$field_name]);
+      // If field is set to reset when untriggered, set it to the default value.
+      if (!empty($field['reset'])) {
+        $entity = $form_state->getFormObject()->getEntity();
+        // If this element is in an inline entity form, we need to get its entity bundle instead of the parent form.
+        if (isset($inline_entity_form_parents)) {
+          $ief = NestedArray::getValue($form, $inline_entity_form_parents['element_parents']);
+          $entity = $ief['#entity'];
+        }
+
+        if ($entity) {
+          $default = $entity->getFieldDefinition($field_name)->getDefaultValue($entity);
+          // Save the default value for the field.
+          NestedArray::setValue($form_state->getValues(), [...$dependent_parents, $field_name], $default);
+        }
+      }
 
       if (!empty($field['errors'])) {
         $untriggered_dependents_errors = array_merge($untriggered_dependents_errors, $field['errors']);
@@ -871,6 +869,13 @@ class ConditionalFieldsFormHelper {
 
       // Reinstate remaining errors.
       foreach ($errors as $name => $error) {
+        // IEF errors bubble up to the save button, so if this error comes from there and
+        // it's already been removed from the originating element, then remove it from the button as well.
+        if (preg_match('/ief_.+_save$/', $name) && in_array($error, $removed_messages)) {
+          $removed_messages[] = $error;
+          continue;
+        }
+
         $form_state->setErrorByName($name, $error);
         // form_set_error() calls drupal_set_message(), so we have to filter out
         // these from the messages to avoid duplicates.
diff --git a/tests/src/FunctionalJavascript/ConditionalFieldUntriggeredTest.php b/tests/src/FunctionalJavascript/ConditionalFieldUntriggeredTest.php
new file mode 100644
index 0000000000000000000000000000000000000000..4ad640d438c8fc8983f8b9245153bf5e4ff9306b
--- /dev/null
+++ b/tests/src/FunctionalJavascript/ConditionalFieldUntriggeredTest.php
@@ -0,0 +1,198 @@
+<?php
+
+namespace Drupal\Tests\conditional_fields\FunctionalJavascript;
+
+use Drupal\Core\Entity\Entity\EntityFormDisplay;
+use Drupal\conditional_fields\ConditionalFieldsInterface;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+
+/**
+ * Test Conditional Fields Text Handler.
+ *
+ * @group conditional_fields
+ */
+class ConditionalFieldUntriggeredTest extends ConditionalFieldTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $screenshotPath = 'sites/simpletest/conditional_fields/untriggered/';
+
+  /**
+   * The test's name to use in file names.
+   *
+   * @var string
+   */
+  protected $testName = 'Untriggered';
+
+  /**
+   * The field name used in the test.
+   *
+   * @var string
+   */
+  protected $fieldName = 'single_textfield';
+
+  /**
+   * The target field name.
+   *
+   * @var string
+   */
+  protected $targetFieldName = 'body';
+
+  /**
+   * The target field wrapper selector.
+   *
+   * @var string
+   */
+  protected $targetFieldWrap = '';
+
+  /**
+   * Jquery selector of field in a document.
+   *
+   * @var string
+   */
+  protected $fieldSelector;
+
+  /**
+   * Jquery selector of target field in a document.
+   *
+   * @var string
+   */
+  protected $targetFieldSelector;
+
+  /**
+   * Default target field value to set in field config.
+   *
+   * @var string
+   */
+  protected $targetFieldDefaultValue;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->fieldSelector = '[name="field_' . $this->fieldName . '[0][value]"]';
+    $this->targetFieldSelector = '[name="' . $this->targetFieldName . '[0][value]"]';
+    $this->targetFieldWrap = '.field--name-' . str_replace('_', '-', $this->targetFieldName);
+    $this->targetFieldDefaultValue = $this->getRandomGenerator()->word(8);
+
+    $fieldStorageDefinition = [
+      'field_name' => 'field_' . $this->fieldName,
+      'entity_type' => 'node',
+      'type' => 'text',
+      'cardinality' => 1,
+    ];
+    $fieldStorage = FieldStorageConfig::create($fieldStorageDefinition);
+    $fieldStorage->save();
+
+    FieldConfig::create([
+      'field_name' => 'field_' . $this->fieldName,
+      'entity_type' => 'node',
+      'bundle' => 'article',
+    ])->save();
+
+    FieldConfig::loadByName('node', 'article', 'body')
+      ->set('default_value', [['value' => $this->targetFieldDefaultValue]])
+      ->save();
+
+    EntityFormDisplay::load('node.article.default')
+      ->setComponent('field_' . $this->fieldName, ['type' => 'text_textfield'])
+      ->save();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function testResetIfUntriggered() {
+
+    $this->baseTestSteps();
+
+    // Visit a ConditionalFields configuration page for Content bundles.
+    $this->createCondition($this->targetFieldName, 'field_' . $this->fieldName, 'unchanged', 'value');
+
+    // Set up conditions.
+    $text = $this->getRandomGenerator()->word(8);
+    $data = [
+      'condition' => 'value',
+      'values_set' => ConditionalFieldsInterface::CONDITIONAL_FIELDS_DEPENDENCY_VALUES_WIDGET,
+      'field_' . $this->fieldName . '[0][value]' => $text,
+      'grouping' => 'AND',
+      'state' => 'unchanged',
+      // Reset if untriggered
+      'reset' => 'true',
+    ];
+
+    $this->createScreenshot($this->screenshotPath . '01-' . $this->testName . __FUNCTION__ . '.png');
+    $this->submitForm($data, 'Save settings');
+
+    // Check if that configuration is saved.
+    $this->drupalGet('admin/structure/types/manage/article/conditionals');
+    $this->assertSession()
+      ->pageTextContains($this->targetFieldName . ' ' . 'field_' . $this->fieldName . ' unchanged value');
+    $this->createScreenshot($this->screenshotPath . '02-' . $this->testName . __FUNCTION__ . '.png');
+
+    // Visit Article Add form to create a test node.
+    $this->drupalGet('node/add/article');
+
+    $this->changeField('[name="title[0][value]"]', 'Test article');
+    // Set the dependency field to something that doesn't trigger the condition.
+    $this->changeField($this->fieldSelector, $text . 'a');
+    // Set the dependent field to something that should be reset since the condition is untriggered.
+    $this->changeField($this->targetFieldSelector, 'this should disappear');
+    $this->createScreenshot($this->screenshotPath . '03-' . $this->testName . __FUNCTION__ . '.png');
+    // Save and create the article
+    $this->submitForm([], 'Save');
+    $this->createScreenshot($this->screenshotPath . '04-' . $this->testName . __FUNCTION__ . '.png');
+    $this->assertSession()->addressMatches('/\/node\/\d+/');
+    $this->assertSession()->pageTextNotContains('this should disappear');
+    $this->assertSession()->pageTextContains($this->targetFieldDefaultValue);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function testDontResetIfUntriggered() {
+
+    $this->baseTestSteps();
+
+    // Visit a ConditionalFields configuration page for Content bundles.
+    $this->createCondition($this->targetFieldName, 'field_' . $this->fieldName, 'unchanged', 'value');
+
+    // Set up conditions.
+    $text = $this->getRandomGenerator()->word(8);
+    $data = [
+      'condition' => 'value',
+      'values_set' => ConditionalFieldsInterface::CONDITIONAL_FIELDS_DEPENDENCY_VALUES_WIDGET,
+      'field_' . $this->fieldName . '[0][value]' => $text,
+      'grouping' => 'AND',
+      'state' => 'unchanged',
+    ];
+
+    $this->createScreenshot($this->screenshotPath . '01-' . $this->testName . __FUNCTION__ . '.png');
+    $this->submitForm($data, 'Save settings');
+
+    // Check if that configuration is saved.
+    $this->drupalGet('admin/structure/types/manage/article/conditionals');
+    $this->assertSession()
+      ->pageTextContains($this->targetFieldName . ' ' . 'field_' . $this->fieldName . ' unchanged value');
+    $this->createScreenshot($this->screenshotPath . '02-' . $this->testName . __FUNCTION__ . '.png');
+
+    // Visit Article Add form to create a test node.
+    $this->drupalGet('node/add/article');
+
+    $this->changeField('[name="title[0][value]"]', 'Test article');
+    // Set the dependency field to something that doesn't trigger the condition.
+    $this->changeField($this->fieldSelector, $text . 'a');
+    // Set the dependent field to something that should not be reset if the condition is untriggered.
+    $this->changeField($this->targetFieldSelector, 'this should still be here');
+    $this->createScreenshot($this->screenshotPath . '03-' . $this->testName . __FUNCTION__ . '.png');
+    // Save and create the article
+    $this->submitForm([], 'Save');
+    $this->createScreenshot($this->screenshotPath . '04-' . $this->testName . __FUNCTION__ . '.png');
+    $this->assertSession()->addressMatches('/\/node\/\d+/');
+    $this->assertSession()->pageTextContains('this should still be here');
+  }
+}
