diff --git a/src/AttachMediaToGroup.php b/src/AttachMediaToGroup.php
index fe41ff2..a6b5872 100644
--- a/src/AttachMediaToGroup.php
+++ b/src/AttachMediaToGroup.php
@@ -12,6 +12,7 @@ use Drupal\group\Entity\GroupContentInterface;
 use Drupal\group\Entity\GroupInterface;
 use Drupal\group\Plugin\GroupContentEnablerManagerInterface;
 use Drupal\media\MediaInterface;
+use Psr\Log\LoggerInterface;
 
 /**
  * Class Attach Media To Group.
@@ -85,7 +86,7 @@ class AttachMediaToGroup {
    * @param \Drupal\Core\Logger\LoggerChannelInterface $logger
    *   The logger channel.
    */
-  public function __construct(MediaFinderManager $mediaFinderManager, GroupContentEnablerManagerInterface $groupEnablerManager, ModuleHandlerInterface $moduleHandler, EntityTypeManagerInterface $entityTypeManager, LoggerChannelInterface $logger) {
+  public function __construct(MediaFinderManager $mediaFinderManager, GroupContentEnablerManagerInterface $groupEnablerManager, ModuleHandlerInterface $moduleHandler, EntityTypeManagerInterface $entityTypeManager, LoggerInterface $logger) {
     $this->mediaFinder = $mediaFinderManager;
     $this->groupEnabler = $groupEnablerManager;
     $this->moduleHandler = $moduleHandler;
