diff --git a/core/assets/scaffold/files/index.php b/core/assets/scaffold/files/index.php
index 750dc282d..e4294dd16 100644
--- a/core/assets/scaffold/files/index.php
+++ b/core/assets/scaffold/files/index.php
@@ -17,6 +17,13 @@
 
 $request = Request::createFromGlobals();
 $response = $kernel->handle($request);
+
+// Hack to allow Auth0 to work with Drupal.
+if (str_contains($request->getPathInfo(), '/auth0')) {
+  require('auth0/index.php');
+  exit;
+}
+
 $response->send();
 
 $kernel->terminate($request, $response);
