diff --git a/js/leaflet.drupal.js b/js/leaflet.drupal.js
index 5401057..b4ff07b 100644
--- a/js/leaflet.drupal.js
+++ b/js/leaflet.drupal.js
@@ -185,9 +185,9 @@
     let map_layer = self.create_layer(definition, key);
     self.base_layers[key] = map_layer;
     // Only the first base layer needs to be added to the map - all the others are accessed via the layer switcher.
-    if (i === 0) {
+    // if (i === 0) {
       self.lMap.addLayer(map_layer);
-    }
+    // }
     if (Drupal.Leaflet[mapid].layer_control == null) {
       self.initialise_layer_control(mapid);
     }
