diff --git a/leaflet.drupal.js b/leaflet.drupal.js
index 09eb000..c041f1e 100644
--- a/leaflet.drupal.js
+++ b/leaflet.drupal.js
@@ -182,9 +182,9 @@
     var map_layer = self.create_layer(definition, key);
     self.base_layers[key] = map_layer;
     // Only the first base layer needs to be added to the map - all the others are accessed via the layer switcher.
-    if (i === 0) {
+    // if (i === 0) {
       self.lMap.addLayer(map_layer);
-    }
+    // }
     if (self.layer_control == null) {
       self.initialise_layer_control();
     }
