diff --git a/js/leaflet.drupal.js b/js/leaflet.drupal.js
index 08192e4..f63af48 100644
--- a/js/leaflet.drupal.js
+++ b/js/leaflet.drupal.js
@@ -296,9 +296,9 @@
     this.base_layers[key] = base_layer;
 
     // Only the first base layer needs to be added to the map - all the others are accessed via the layer switcher.
-    if (i === 0) {
+    // if (i === 0) {
       this.lMap.addLayer(base_layer);
-    }
+    // }
 
     // Initialise the Layers Control, if not yet.
     if (this.layer_control == null) {
