diff --git a/src/Controller/ListUsageController.php b/src/Controller/ListUsageController.php
index 065612f..43084eb 100644
--- a/src/Controller/ListUsageController.php
+++ b/src/Controller/ListUsageController.php
@@ -292,7 +292,11 @@ class ListUsageController extends ControllerBase {
           }
         }
 
-        $field_definitions = $this->entityFieldManager->getFieldDefinitions($source_type, $source_entity->bundle());
+        try {
+          $field_definitions = $this->entityFieldManager->getFieldDefinitions($source_type, $source_entity->bundle());
+        } catch (\Exception $e) {
+          $field_definitions = [];
+        }
         $default_langcode = $source_entity->language()->getId();
         $used_in = [];
         $revisions = [];
@@ -323,7 +327,16 @@ class ListUsageController extends ControllerBase {
               '#list_type' => 'ul',
             ];
           }
-        }
+        } else {
+          // If the entity is not revisionable, just pick the first record.
+          if (array_key_exists(0, $records)) {
+            [
+              'source_langcode' => $source_langcode,
+              'field_name' => $field_name,
+            ] = $records[0];
+            $revisions[static::REVISION_DEFAULT][$source_langcode] = $field_name;
+          }
+         }
         $link = $this->getSourceEntityLink($source_entity);
         // If the label is empty it means this usage shouldn't be shown
         // on the UI, just skip this row.
@@ -339,7 +352,7 @@ class ListUsageController extends ControllerBase {
           }
         };
         $field_name = $get_field_name($revisions);
-        $field_label = isset($field_definitions[$field_name]) ? $field_definitions[$field_name]->getLabel() : $this->t('Unknown');
+        $field_label = isset($field_definitions[$field_name]) ? $field_definitions[$field_name]->getLabel() : $field_name ?? $this->t('Unknown');
         $type = $entity_types[$source_type]->getLabel();
         if ($source_bundle_key = $source_entity->getEntityType()->getKey('bundle')) {
           $bundle_field = $source_entity->{$source_bundle_key};
