diff --git a/js/Leaflet.ResetView/dist/L.Control.ResetView.js b/js/Leaflet.ResetView/dist/L.Control.ResetView.js
index f8f9fcc9..ecd0ac97 100644
--- a/js/Leaflet.ResetView/dist/L.Control.ResetView.js
+++ b/js/Leaflet.ResetView/dist/L.Control.ResetView.js
@@ -56,7 +56,7 @@
       // Added by itamair according to this drustack/Leaflet.ResetView issue:
       // https://github.com/drustack/Leaflet.ResetView/issues/1
       if (e) { L.DomEvent.preventDefault(e) }
-      this._map.setView(this.options.latlng, this.options.zoom);
+      this._map.setView(this._map.options.initCenter, this._map.options.initZoom);
     },
   });
 
