diff --git a/src/EasyBreadcrumbBuilder.php b/src/EasyBreadcrumbBuilder.php
index 304d57e..bdff308 100644
--- a/src/EasyBreadcrumbBuilder.php
+++ b/src/EasyBreadcrumbBuilder.php
@@ -239,6 +239,10 @@ class EasyBreadcrumbBuilder implements BreadcrumbBuilderInterface {
               $title = $replacedTitles[$title];
             }
           }
+          // If this is a route to taxonomy term, set the breadcrumb title to term name. 
+          if($route_match->getParameter('taxonomy_term') != NULL){
+            $title = $route_match->getParameter('taxonomy_term')->getName();
+          }
           if (!isset($title)) {
 
             if ($this->config->get(EasyBreadcrumbConstants::USE_MENU_TITLE_AS_FALLBACK)) {
