diff --git a/src/Plugin/search_api/backend/SearchApiSolrBackend.php b/src/Plugin/search_api/backend/SearchApiSolrBackend.php
index b880a36..eb7be5b 100644
--- a/src/Plugin/search_api/backend/SearchApiSolrBackend.php
+++ b/src/Plugin/search_api/backend/SearchApiSolrBackend.php
@@ -1127,7 +1127,12 @@ class SearchApiSolrBackend extends BackendPluginBase implements SolrBackendInter
           }
           else {
             if ($field->getDataDefinition()->isList() || $this->isHierarchicalField($field)) {
-              $pref .= 'm';
+              if ($field->getPropertyPath() == 'aggregated_field' && $field->getConfiguration()['type'] != 'union') {
+                $pref .= 's';
+              }
+              else {
+                $pref .= 'm';
+              }
             }
             else {
               try {
