diff --git a/modules/geocoder_field/geocoder_field.module b/modules/geocoder_field/geocoder_field.module
index e699a4a..55fef99 100644
--- a/modules/geocoder_field/geocoder_field.module
+++ b/modules/geocoder_field/geocoder_field.module
@@ -194,7 +194,7 @@ function geocoder_field_entity_presave(EntityInterface $entity) {
       $entity->get($geocoder['field']) :
       NULL;
 
-    if ($remote_field === NULL) {
+    if ($remote_field === NULL || $remote_field->first()->address_line1 === NULL) {
       continue;
     }
 
