From a2f74016019e6f1ed7e56bac4ab9aebfa8a47ffe Mon Sep 17 00:00:00 2001
From: Oden <hello@odensc.com>
Date: Thu, 16 Apr 2026 15:56:26 -0700
Subject: [PATCH] Fix disappearing AJAX elements when using "invisible" state

---
 src/WebformSubmissionConditionsValidator.php | 21 ++++++++++----------
 1 file changed, 10 insertions(+), 11 deletions(-)

diff --git a/src/WebformSubmissionConditionsValidator.php b/src/WebformSubmissionConditionsValidator.php
index cb153ac..c297bcd 100644
--- a/src/WebformSubmissionConditionsValidator.php
+++ b/src/WebformSubmissionConditionsValidator.php
@@ -124,17 +124,6 @@ class WebformSubmissionConditionsValidator implements WebformSubmissionCondition
         $all_targets_visible = (array_sum($targets) === count($targets));
         $has_cross_page_targets = (!$all_targets_visible && array_sum($targets));

-        // Skip if evaluating conditions when all targets are visible.
-        if ($all_targets_visible) {
-          // Add .js-webform-states-hidden to element's that are not visible when
-          // the form is rendered.
-          if (str_starts_with($state, 'visible')
-            && !$this->validateConditions($conditions, $webform_submission)) {
-            $this->addStatesHiddenToElement($element);
-          }
-          continue;
-        }
-
         // Replace hidden cross page targets with hidden inputs.
         if ($has_cross_page_targets) {
           $cross_page_targets = array_filter(
@@ -157,6 +146,16 @@ class WebformSubmissionConditionsValidator implements WebformSubmissionCondition
         // Negate the result.
         $result = ($negate) ? !$result : $result;

+        // Skip removing the state and clearing the value from elements that are not visible
+        // when the form is rendered, if all of its targets are present on the current page.
+        // Instead, just add the add the .js-webform-states-hidden class.
+        // This way visiblity can be re-computed in the browser based on later user input.
+        if ($all_targets_visible && str_starts_with($state, 'visible') && !$result) {
+          $this->addStatesHiddenToElement($element);
+
+          continue;
+        }
+
         // Apply result to element state.
         switch ($state) {
           case 'required':
--
2.34.1

